### PR TITLE
new definition file for hls.js

### PR DESCRIPTION
**Pull request by jgainfort**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/hls.js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,1261 @@
...
+   * hls.js event listener
+   */
+  on(event: string, callback: (event: string, data: Hls.Data) => void): void;
+}
+
+declare var Hls: HlsFactory;
+export as namespace Hls;
+export = Hls;
+
+// TypeScript Version: 2.2
```
Comment (andy-ms) [inline types/hls.js/index.d.ts]: Put this on line 5.

**Review by ghost — COMMENTED** [review]
Instead of using `HlsFactory`, try structuring it with a class like this:

```ts
declare namespace Hls {
    ...
    namespace Events {
        const MEDIA_ATTACHING: string;
        ...
    }
    ...
    const version: number;
}

declare class Hls {
    constructor(config?: Hls.Config);
    ...
}

export as namespace Hls;
export = Hls;
```

**Comment by jgainfort** [comment]
@andy-ms makes sense I can do that. Just one question. How do I handle the DefaultConfig namespace in the Hls namespace?

Previously in the factory I just had DefaultConfig: Hls.Config

```
namespace Hls {
  ...
  namespace DefaultConfig {
    ...
  }
  ...
  export interface Config {
    ...
  }
}
```

my first though was to just put the contents of the Config interface into it but then I would have duplications of the properties of the interface within the namespace.

**Comment by jgainfort** [comment]
@andy-ms next thought is to just do

```
namespace Hls {
  ...
  let DefaultConfig: Config;
  ...
  export interface Config {
    ...
  }
}
```

**Comment by andy-ms** [comment]
Yeah, don't duplicate. Be sure to use `let` only if it's meant to be mutated, else use `const`.

**Comment by jgainfort** [comment]
Cool, thank you. It is meant to be duplicated.

**Comment by andy-ms** [comment]
Thanks!

**Comment by jgainfort** [comment]
@andy-ms I cant find this package on npm? When are the packages updated?

**Comment by andy-ms** [comment]
It's up now.

**Comment by jgainfort** [comment]
Nice! Thank you.

**Pull request merged**